File: projects/gardens/src/java/com/threerings/gardens/logic/browse.java
//
// $Id$

package com.threerings.gardens.logic;

import java.util.List;
import java.util.Collections;
import java.util.Comparator;

import com.samskivert.servlet.user.User;
import com.samskivert.servlet.util.ParameterUtil;
import com.samskivert.velocity.InvocationContext;

import com.threerings.toybox.server.persist.GameRecord;

import com.threerings.gardens.Log;
import com.threerings.gardens.GardensApp;

/**
 * Displays a list of all registered games.
 */
public class browse extends OptionalUserLogic
{
    // documentation inherited
    public void invoke (InvocationContext ctx, GardensApp app, User user)
        throws Exception
    {
        String category = ParameterUtil.getParameter(
            ctx.getRequest(), "category", false);

        List<GameRecord> games = category.equals("") ?
            // load up the metadata for all of our games
            app.getToyBoxRepository().loadGames() :
            // load up the metadata for the games in this category
            app.getToyBoxRepository().loadGames(category);

        // sort our games by name
        Collections.sort(games, new Comparator<GameRecord>() {
            public int compare (GameRecord g1, GameRecord g2) {
                return g1.name.compareTo(g2.name);
            }
        });
        ctx.put("games", games);
        ctx.put("category", category);
    }
}

File: server/src/main/scala/com/threerings/gardens/web/logic/browse.scala
//
// Game Gardens - a platform for hosting simple multiplayer Java games
// Copyright (c) 2005-2013, Three Rings Design, Inc. - All rights reserved.
// https://github.com/threerings/game-gardens/blob/master/LICENSE

package com.threerings.gardens.web.logic

import java.util.{List, Collections, Comparator}

import com.samskivert.servlet.user.User
import com.samskivert.servlet.util.ParameterUtil
import com.samskivert.velocity.InvocationContext

import com.threerings.toybox.server.persist.GameRecord

import com.threerings.gardens.web.GardensApp

/** Displays a list of all registered games. */
class browse extends OptionalUserLogic {

  override def invoke (ctx :InvocationContext, app :GardensApp, user :User) {
    val category = ParameterUtil.getParameter(ctx.getRequest, "category", false)

    // load up the metadata for all of our games
    val games = if (category.equals("")) app.toyBoxRepo.loadGames()
    // load up the metadata for the games in this category
    else app.toyBoxRepo.loadGames(category)

    // sort our games by name
    Collections.sort(games, new Comparator[GameRecord]() {
      def compare (g1 :GameRecord, g2 :GameRecord) = g1.name.compareTo(g2.name)
    })
    ctx.put("games", games)
    ctx.put("category", category)
  }
}
